packages/protocol-kit/src/Safe.ts

```diff
@@ -602,6 +605,34 @@ class Safe {
    * @returns The Safe signature
    */
   async signHash(hash: string): Promise<SafeSignature> {
+    const isPasskeySigner = await this.#safeProvider.isPasskeySigner()
```

Comment:
All this logic make sense in an utility function right? same as generateSignature